File: chain-sdk/clojure/src/on_chain_api/specs/token_v2_dto.clj
(ns on-chain-api.specs.token-v2-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            )
  (:import (java.io File)))


(def token-v2-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :symbol) string?
   (ds/opt :name) string?
   (ds/opt :decimals) int?
   (ds/opt :total_supply) big-integer-spec
   (ds/opt :trade_volume) string?
   (ds/opt :trade_volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :tx_count) big-integer-spec
   (ds/opt :total_liquidity) string?
   (ds/opt :derived_eth) string?
   (ds/opt :token_symbol) string?
   })

(def token-v2-dto-spec
  (ds/spec
    {:name ::token-v2-dto
     :spec token-v2-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/TokenV2DTO.js
goog.provide('API.Client.TokenV2DTO');

/**
 * Stores aggregated information for a specific token across all pairs that token is included in.
 * @record
 */
API.Client.TokenV2DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV2DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV2DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.TokenV2DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenV2DTO.prototype.vid;

/**
 * Token address.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.id;

/**
 * Token symbol.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.symbol;

/**
 * Token name.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.name;

/**
 * Token decimals.
 * @type {!number}
 * @export
 */
API.Client.TokenV2DTO.prototype.decimals;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TokenV2DTO.prototype.totalSupply;

/**
 * Amount of token traded all time across all pairs.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.tradeVolume;

/**
 * Amount of token in USD traded all time across pairs (only for tokens with liquidity above minimum threshold).
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.tradeVolumeUsd;

/**
 * Amount of token in USD traded all time across pairs (no minimum liquidity threshold).
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.untrackedVolumeUsd;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TokenV2DTO.prototype.txCount;

/**
 * Total amount of token provided as liquidity across all pairs.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.totalLiquidity;

/**
 * ETH per token.
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.derivedEth;

/**
 * @type {!string}
 * @export
 */
API.Client.TokenV2DTO.prototype.tokenSymbol;

